src/Storages/ObjectStorageQueue/ObjectStorageQueueSource.cpp

```diff
@@ -593,6 +593,20 @@ ObjectStorageQueueSource::FileIterator::getNextKeyFromAcquiredBucket(size_t proc
         if (current_bucket_holder && current_bucket_holder->isZooKeeperSessionExpired())
         {
             LOG_TRACE(log, "ZooKeeper session expired, bucket {} not longer hold", current_bucket_holder->getBucket());
+
+            for (auto & [bucket, bucket_info] : listed_keys_cache)
+            {
+                /// Reset current processor for the keys
+                /// to avoid the above error "Expected current processor {} to be equal to {} for bucket {}".
+                /// TODO: Support persistent bucket holder keeper node,
+                /// like since recently we support persistent processing nodes.
```

Comment:
```suggestion
                /// similar to support for persistent processing nodes.
```

"recently" is relative and will become false quite soon 🙂

Reply:
TODO is resolved here https://github.com/ClickHouse/ClickHouse/pull/86628